entity-service/src/main/kotlin/com/egm/stellio/entity/util/EntitiesGraphBuilder.kt

```diff
@@ -0,0 +1,78 @@
+package com.egm.stellio.entity.util
+
+import com.egm.stellio.entity.repository.Neo4jRepository
+import com.egm.stellio.entity.web.BatchEntityError
+import com.egm.stellio.shared.model.BadRequestDataException
+import com.egm.stellio.shared.model.ExpandedEntity
+import org.jgrapht.Graph
+import org.jgrapht.graph.DefaultEdge
+import org.jgrapht.graph.DirectedPseudograph
+import org.springframework.stereotype.Component
+
+/**
+ * Builder for [Graph].
+ */
+@Component
+class EntitiesGraphBuilder(
+    private val neo4jRepository: Neo4jRepository
+) {
+
+    /**
+     * Builds a graph based on given [entities].
+     *
+     * @return a [Pair] :  a [Graph] of [ExpandedEntity] and a [List] of [BatchEntityError]
+     */
```

Comment:
Don't feel forced to add javadoc if you have nothing special to say. For instance, the @return doesn't say more than what is in the method signature.

Reply:
Yes indeed I've modified that to add information about what's returned (mentionning the filtering of entities without valid linked entities)